libevmasm/GasMeter.h

```diff
@@ -47,11 +53,28 @@ class GasMeter
 		bool isInfinite;
 	};
 
-	/// Returns an upper bound on the gas consumed by the given instruction.
+	/// Constructs a new gas meter given the current state.
+	GasMeter(std::shared_ptr<KnownState> const& _state): m_state(_state) {}
+
+	/// @returns an upper bound on the gas consumed by the given instruction and updates
+	/// the state.
 	GasConsumption estimateMax(AssemblyItem const& _item);
 
 private:
+	/// @returns _multiplier * (_value + 31) / 32, if _value is a known constant and infinite otherwise.
+	GasConsumption wordGas(u256 const& _multiplier, ExpressionClasses::Id _value);
```

Comment:
No, they cannot. They query expression classes and any query to expression classes might create a new class.